src/core/markovProcessGraph.js

```diff
@@ -644,33 +644,64 @@ export class MarkovProcessGraph {
     return this._participants;
   }
 
+  /**
+   * Returns a canonical ordering of the nodes in the graph.
+   *
+   * No assumptions should be made about the node order, other than
+   * that it is stable for any given MarkovProcessGraph.
+   */
+  nodeOrder(): $ReadOnlyArray<NodeAddressT> {
+    return Array.from(this._nodes.keys()).sort();
```

Comment:
Can we make `this._nodes`’s insertion order stable and just `yield*
this._nodes.keys()`, returning an iterator with constant time and
constant per-element advancement rather than linearithmic upfront time?